### PR TITLE
feat(core): support styles and styleUrl as strings

### PR DESCRIPTION
**Pull request by crisbeto**
Adds support for passing in `@Component.styles` as a string. Also introduces a new `styleUrl` property on `@Component` for providing a single stylesheet. This is more convenient for the most common case where a component only has one stylesheet associated with it.

### CODE REVIEW

**Comment by eneajaho** [comment]
Hello @crisbeto 

This example below looks like it accepts multiple urls because of the naming. 
`styleUrls: 'dir/styles.css, dir/styles1.css'`

Imo, The correct way it should look like is: 
`styleUrl: 'dir/styles.css'`

But that would require big changes I believe.

**Comment by crisbeto** [comment]
Yeah, I agree that the naming is problematic, but I think that splitting it on the comma would be weird as well. Introducing a `styleUrl` property won't be that much more work to support, but it increases the API surface. WDYT @jelbourn?

**Comment by jnizet** [comment]
My two cents here: the Angular API already has at least one other place where it uses the plural form to also accept a single thing: https://angular.io/api/forms/AbstractControlOptions. So keeping `styleUrls` would at least be consistent.

**Comment by crisbeto** [comment]
Based on what we discussed with @jelbourn: I've introduced a new `styleUrl` property and kept `styleUrls` as a `string[]`.

File: packages/core/test/metadata/resource_loading_spec.ts [hunk 1]
```diff
@@ -122,6 +123,49 @@ Did you run and wait for 'resolveComponentResources()'?`.trim());
...
+      const metadata: Component = {template: '', styleUrl: 'test://style', styles: 'existing'};
+      compileComponent(MyComponent, metadata);
+      await resolveComponentResources(testResolver);
+      expect(MyComponent.ɵcmp).toBeDefined();
+      expect(metadata.styleUrls).toBe(undefined);
+      expect(metadata.styles).toEqual(['existing', 'style']);
+      expect(resourceFetchCount).toBe(1);
+    });
+
+    it('should throw if both styles and styleUrl are passed in', async () => {
```
Comment (pkozlowski-opensource) [inline packages/core/test/metadata/resource_loading_spec.ts:160]: test name: `should throw if both styleUrls and styleUrl are passed in`

**Review by pkozlowski-opensource — APPROVED** [review]
LGTM

Reviewed-for: public-api 
Reviewed-for: fw-core

**Review by jessicajaniuk — APPROVED** [review]

**Review by jessicajaniuk — APPROVED** [review]
reviewed-for: fw-core, fw-testing, public-api

**Comment by AndrewKushnir** [comment]
This PR was merged into the repository by commit 59387ee476dff1a893a01fe5cbee3c95b93c0cdb.

**Comment by spock123** [comment]
Thank you guys, I know this is a silly small thing but it's really something I bet all developers have come across: it really makes the developing experience better, because 99% of all cases there'll only be 1 styles file. Thanks!!!

**Pull request closed without merging**